cli/src/trusted_commands.rs

```diff
@@ -0,0 +1,243 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{
+	trusted_command_utils::{
+		get_accountid_from_str, get_identifiers, get_keystore_path, get_pair_from_str,
+	},
+	trusted_operation::perform_trusted_operation,
+	Cli,
+};
+use codec::Decode;
+use ita_stf::{Index, KeyPair, TrustedCall, TrustedGetter, TrustedOperation};
+use log::*;
+use my_node_runtime::Balance;
+use sp_application_crypto::{ed25519, sr25519};
+use sp_core::{crypto::Ss58Codec, sr25519 as sr25519_core, Pair};
+use substrate_client_keystore::{KeystoreExt, LocalKeystore};
+
+macro_rules! get_layer_two_nonce {
+	($signer_pair:ident, $cli: ident, $trusted_args:ident ) => {{
+		let top: TrustedOperation =
+			TrustedGetter::nonce(sr25519_core::Public::from($signer_pair.public()).into())
+				.sign(&KeyPair::Sr25519($signer_pair.clone()))
+				.into();
+		let res = perform_operation($cli, $trusted_args, &top);
+		let nonce: Index = if let Some(n) = res {
+			if let Ok(nonce) = Index::decode(&mut n.as_slice()) {
+				nonce
+			} else {
+				0
+			}
+		} else {
+			0
+		};
+		debug!("got layer two nonce: {:?}", nonce);
+		nonce
+	}};
+}
+
+#[derive(Args)]
+pub struct TrustedArgs {
+	/// targeted worker MRENCLAVE
+	#[clap(short, long)]
+	pub(crate) mrenclave: String,
+
+	/// shard identifier
+	#[clap(short, long)]
+	pub(crate) shard: Option<String>,
+
+	/// signer for publicly observable extrinsic
+	#[clap(short='a', long, default_value_t = String::from("//Alice"))]
+	pub(crate) xt_signer: String,
+
+	/// insert if direct invocation call is desired
+	#[clap(short, long)]
+	direct: bool,
+
+	#[clap(subcommand)]
+	command: TrustedCommands,
+}
+
+#[derive(Subcommand)]
+pub enum TrustedCommands {
+	/// generates a new incognito account for the given integritee shard
+	NewAccount,
+
```

Comment:
I would not call this `integritee shard`, maybe just shard? open to suggestions.